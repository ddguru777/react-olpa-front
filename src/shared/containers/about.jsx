/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { appSetTitle } from "zoapp-front/actions/app";

const aboutStyle = {
  fontSize: "18px",
  fontWeight: "300",
  color: "#666",
  margin: "36px",
};
const infoStyle = {
  fontSize: "24px",
  fontWeight: "300",
  color: "#666",
  paddingBottom: "20px",
};

export class About extends Component {
  componentWillMount() {
    this.props.appSetTitle("About");
  }

  render() {
    const { isLoading } = this.props;

    return (
      <div style={aboutStyle}>
        <div style={infoStyle}>{isLoading ? "" : null} </div>
        <p>
          We are Opla, the mightly volcanic team ! Since 2015 we started
          developing this deeptech chatbot solution. We are truly commited to
          OpenSource. And we are fully involved in creating the best and
          complete tool for conversational interface, user oriented.
        </p>
        <p>
          Version<br />0.1 : Amazing Alou (Le Puy Alou : a french volcano) June
          2018<br />
          <a
            href="https://github.com/Opla/"
            target="_blank"
            rel="noreferrer noopener"
          >
            See more on Github
          </a>
        </p>
        <p>
          Thanks to the team and contributors<br />
          Julie, Maali, Marie, Nicolle, Dorian, Loic, Manu, Mik, Thomas, Will,
          Will
        </p>
        <p>
          Follow us on Twitter @opla_ai<br />
          Or contact us hello@opla.ai
        </p>
      </div>
    );
  }
}

About.propTypes = {
  isLoading: PropTypes.bool,
  appSetTitle: PropTypes.func.isRequired,
};
const mapDispatchToProps = (dispatch) => ({
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
});

export default connect(null, mapDispatchToProps)(About);
