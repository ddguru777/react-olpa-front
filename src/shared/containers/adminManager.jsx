/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import Zrmc, {
  Grid,
  Inner,
  Cell,
  Button,
  Icon,
  TextField,
  Select,
  MenuItem,
} from "zrmc";
import { TableComponent } from "zoapp-ui";
import { connect } from "react-redux";
import Loading from "zoapp-front/components/loading";
import SignInForm from "zoapp-front/containers/signInForm";
import { apiSetAdminParametersRequest } from "zoapp-front/actions/api";

import { appSetTitle } from "../actions/app";
import { apiSaveBotRequest } from "../actions/api";
import ServicesContainer from "./servicesContainer";
import PluginsManager from "../utils/pluginsManager";
import TunnelBox from "../components/tunnelBox";
import timezones from "../utils/timezones";

const infoStyleD = {
  fontSize: "16px",
  fontWeight: "400",
  color: "#666",
  padding: "24px",
  lineHeight: "1.1",
};

const FORM_WIDTH = "100%";

class AdminManager extends Component {
  static onActionTunnel(/* dialog, action */) {
    Zrmc.closeDialog();
  }

  constructor(props) {
    super(props);
    const pluginsManager = new PluginsManager();

    this.state = {
      pluginsManager,
      bot: props.bot,
      tunnelParams: null,
      backendParams: null,
      emailServerParams: null,
    };
  }

  componentWillMount() {
    this.props.appSetTitle("Admin");
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.bot !== this.props.bot) {
      this.setState({
        bot: nextProps.bot,
      });
    }
  }

  onChangeTunnel = (tunnelParams) => {
    this.setState({ tunnelParams });
  };

  onSaveBotDetails = () => {
    this.props.apiSaveBotRequest(this.state.bot);
  };

  onSaveBackend() {
    if (this.state.tunnelParams) {
      // WIP save tunnel parameters
      const tunnel = this.state.tunnelParams.active || "None";
      this.setState({ tunnelParams: null });
      this.props.apiSetAdminParametersRequest({ tunnel });
    }
  }

  displayTunnelDialog() {
    const { params } = this.props.admin;
    const backend = this.state.backendParams || params.backend || {};
    const tunnelParams = this.state.tunnelParams || backend.tunnel || {};
    const body = (
      <TunnelBox onChange={this.onChangeTunnel} params={tunnelParams} />
    );
    Zrmc.showDialog({
      header: "Tunnel settings",
      body,
      syle: { width: "520px" },
      onAction: AdminManager.onActionTunnel,
    });
  }

  handleBotNameChange = (e) => {
    const name = e.target.value;

    this.setState({
      bot: {
        ...this.state.bot,
        name,
      },
    });
  };

  handleBotDescriptionChange = (e) => {
    const description = e.target.value;

    this.setState({
      bot: {
        ...this.state.bot,
        description,
      },
    });
  };

  handleLanguageChange = (language) => {
    this.setState({
      bot: {
        ...this.state.bot,
        language,
      },
    });
  };

  handleTimezoneChange = (timezone) => {
    this.setState({
      bot: {
        ...this.state.bot,
        timezone,
      },
    });
  };

  render() {
    let { isLoading } = this.props;
    if (!isLoading && !this.props.admin && this.props.isSignedIn) {
      isLoading = true;
    }
    if (!this.props.isSignedIn) {
      return <SignInForm />;
    } else if (isLoading || this.props.admin == null) {
      return <Loading />;
    }
    const active = this.props.activeTab;
    let content = "";
    if (active === 0) {
      content = (
        <Grid>
          <Inner>
            <Cell
              className="mdl-color--white"
              span={12}
              style={{ display: "table" }}
            >
              <div style={{ width: "200px", display: "table-cell" }}>
                <div
                  style={{
                    position: "absolute",
                    width: "180px",
                    height: "180px",
                    margin: "24px",
                    backgroundColor: "#ddd",
                  }}
                />
              </div>
              <div style={{ display: "table-cell" }}>
                <form style={infoStyleD}>
                  <div>
                    <TextField
                      defaultValue={this.props.bot.name}
                      label="Assistant name"
                      onChange={this.handleBotNameChange}
                      style={{ width: FORM_WIDTH }}
                    />
                  </div>
                  <div>
                    <TextField
                      defaultValue={this.props.bot.description}
                      isTextarea
                      label="Description"
                      onChange={this.handleBotDescriptionChange}
                      style={{ width: FORM_WIDTH }}
                    />
                  </div>
                  <div>
                    <Select
                      label="Language"
                      onSelected={this.handleLanguageChange}
                      style={{ width: FORM_WIDTH }}
                      selectedIndex={["en", "fr"].findIndex(
                        (language) =>
                          language === (this.props.bot.language || null),
                      )}
                    >
                      <MenuItem value="en">English</MenuItem>
                      <MenuItem value="fr">French</MenuItem>
                    </Select>
                  </div>
                  <div>
                    <Select
                      label="Timezone"
                      onSelected={this.handleTimezoneChange}
                      style={{ width: FORM_WIDTH }}
                      selectedIndex={this.props.timezones.findIndex(
                        (tz) => tz === (this.props.bot.timezone || null),
                      )}
                    >
                      {this.props.timezones.map((timezone) => (
                        <MenuItem key={timezone} value={timezone}>
                          {timezone}
                        </MenuItem>
                      ))}
                    </Select>
                  </div>
                </form>
              </div>
              <div>
                <Button
                  disabled={
                    !this.state.bot ||
                    (this.state.bot && this.state.bot.name.length < 1)
                  }
                  raised
                  style={{ float: "right", margin: "24px" }}
                  onClick={this.onSaveBotDetails}
                >
                  SAVE
                </Button>
              </div>
            </Cell>
          </Inner>
        </Grid>
      );
    } else if (active === 1) {
      content = (
        <Grid>
          <ServicesContainer pluginsManager={this.state.pluginsManager} />
        </Grid>
      );
    } else if (active === 2) {
      const items = [];
      const status = "you";
      const { user, profile } = this.props;

      const values = [];
      values.push(profile.username);
      values.push(profile.email);
      values.push(user.attributes.scope);
      values.push(status);

      items.push({ id: 1, values, icon: `../images/${profile.avatar}.png` });

      const headers = ["", "username", "email", "role", "status"];
      const title = (
        <div style={infoStyleD}>
          You could give an access to your collaborators here.
          <Button raised style={{ float: "right", marginBottom: "24px" }}>
            ADD
          </Button>
        </div>
      );

      content = (
        <Grid>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <div>
                <TableComponent
                  title={title}
                  headers={headers}
                  items={items}
                  selectedItem={-1}
                  onSelect={() => {}}
                />
              </div>
            </Cell>
          </Inner>
        </Grid>
      );
    } else if (active === 3) {
      const { params } = this.props.admin;

      const emailServer = this.state.emailParams || params.emailServer || {};
      const backend = this.state.backendParams || params.backend || {};
      // const tunnelParams = this.state.tunnelParams || backend.tunnel || {};
      /* const hasTunnelParams = !!this.state.tunnelParams; */
      const saveBackendDisabled = !(
        this.state.backendParams || this.state.tunnelParams
      );
      const saveEmailDisabled = !this.state.emailServerParams;
      content = (
        <Grid>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <div style={infoStyleD}>
                Backend configuration
                <Button
                  raised
                  disabled={saveBackendDisabled}
                  style={{ float: "right" }}
                  onClick={(e) => {
                    e.preventDefault();
                    this.onSaveBackend();
                  }}
                >
                  SAVE
                </Button>
              </div>
              <form style={infoStyleD}>
                <div style={{ width: "520px" }}>
                  <TextField
                    onChange={() => {}}
                    label="Public Api url"
                    style={{ width: FORM_WIDTH }}
                    value={backend.publicUrl}
                  />
                  <Icon
                    /* colored={hasTunnelParams} */
                    style={{ float: "right", marginTop: "8px" }}
                    name="link"
                    onClick={(e) => {
                      e.preventDefault();
                      this.displayTunnelDialog();
                    }}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Api url"
                    disabled
                    style={{ width: FORM_WIDTH }}
                    value={backend.apiUrl}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Auth url"
                    disabled
                    style={{ width: FORM_WIDTH }}
                    value={backend.authUrl}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="AppId"
                    disabled
                    style={{ width: FORM_WIDTH }}
                    value={backend.clientId}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Secret"
                    disabled
                    style={{ width: FORM_WIDTH }}
                    value={backend.clientSecret}
                  />
                </div>
              </form>
              <div />
            </Cell>
          </Inner>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <div style={infoStyleD}>
                Email server configuration
                <Button
                  raised
                  disabled={saveEmailDisabled}
                  style={{ float: "right" }}
                >
                  SAVE
                </Button>
              </div>
              <form style={infoStyleD} autoComplete="nope">
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Server address"
                    style={{ width: FORM_WIDTH }}
                    value={emailServer.url}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Username"
                    autoComplete="new-password"
                    style={{ width: FORM_WIDTH }}
                    value={emailServer.username}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Password"
                    autoComplete="new-password"
                    type="password"
                    style={{ width: FORM_WIDTH }}
                    value={emailServer.password}
                  />
                </div>
              </form>
              <div />
            </Cell>
          </Inner>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <div style={infoStyleD}>
                <span style={{ color: "#d50000" }}>Delete this assistant</span>
                <Button
                  raised
                  style={{
                    float: "right",
                    marginBottom: "24px",
                    backgroundColor: "#d50000",
                  }}
                >
                  DELETE
                </Button>
              </div>
              <div />
            </Cell>
          </Inner>
        </Grid>
      );
    }
    return (
      <div className="mdl-layout__content mdl-color--grey-100">
        <section>{content}</section>
      </div>
    );
  }
}

AdminManager.defaultProps = {
  activeTab: 0,
  admin: null,
  bot: null,
  isLoading: false,
  isSignedIn: false,
  profile: {},
  timezones,
  user: null,
};

AdminManager.propTypes = {
  activeTab: PropTypes.number,
  admin: PropTypes.shape({ params: PropTypes.shape({}).isRequired }),
  apiSaveBotRequest: PropTypes.func.isRequired,
  apiSetAdminParametersRequest: PropTypes.func.isRequired,
  appSetTitle: PropTypes.func.isRequired,
  bot: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    language: PropTypes.string,
    timezone: PropTypes.string,
  }),
  isLoading: PropTypes.bool,
  isSignedIn: PropTypes.bool,
  profile: PropTypes.shape({}),
  timezones: PropTypes.arrayOf(PropTypes.string).isRequired,
  user: PropTypes.shape({}),
};

const mapStateToProps = (state) => {
  const { admin } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.loading;
  const selectedBotId = state.app ? state.app.selectedBotId : null;
  const { user } = state;
  const profile = user ? user.profile : null;

  // TODO get selectedBot from selectBotId
  const bot = selectedBotId ? admin.bots[0] : null;

  return {
    admin,
    isLoading,
    isSignedIn,
    selectedBotId,
    bot,
    user,
    profile,
  };
};

const mapDispatchToProps = (dispatch) => ({
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
  apiSaveBotRequest: (params) => {
    dispatch(apiSaveBotRequest(params));
  },
  apiSetAdminParametersRequest: (params) => {
    dispatch(apiSetAdminParametersRequest(params));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(AdminManager);
