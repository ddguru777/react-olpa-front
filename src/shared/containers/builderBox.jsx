import React, { Component } from "react";
import PropTypes from "prop-types";
import { Button } from "react-mdl";
import { DialogManager } from "zoapp-ui";
import { connect } from "react-redux";
import PublishDialog from "./publishDialog";

class BuilderBox extends Component {
  constructor(props) {
    super(props);
    this.state = { };
  }

  onAction = (action) => {
    console.log("TODO Publish");
    if (action === "Publish") {
      return false;
    }
    return true;
  }

  handleOpenPublishDialog = () => {
    const dialog = <PublishDialog open onAction={this.onAction} store={this.props.store} />;
    DialogManager.open({ dialog });
  }

  render() {
    if (this.props.isSignedIn) {
      const style = { paddingRight: "20%" };
      return (
        <div style={style}>
          <Button
            raised
            colored
            onClick={(e) => { e.preventDefault(); this.handleOpenPublishDialog(); }}
          >
          Publish
          </Button>
        </div>);
    }
    return (<div />);
  }
}

BuilderBox.defaultProps = {
  isSignedIn: false,
  store: null,
};

BuilderBox.propTypes = {
  isSignedIn: PropTypes.bool,
  store: PropTypes.shape({}),
};

const mapStateToProps = (state) => {
  const selectedBotId = state.app ? state.app.selectedBotId : null;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.app.loading;
  return { selectedBotId, isSignedIn, isLoading };
};

export default connect(mapStateToProps)(BuilderBox);
