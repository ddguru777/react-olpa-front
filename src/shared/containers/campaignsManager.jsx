import React, { Component } from "react";
import PropTypes from "prop-types";
import { Content } from "react-mdl";
import { connect } from "react-redux";
import { appSetTitle } from "../actions/app";


class CampaignsManager extends Component {
  componentWillMount() {
    this.props.appSetTitle("Campaigns");
  }

  render() {
    return (
      <Content className="mdl-color--grey-100">
        <section className="text-section" style={{ margin: "40px" }}>
          <div>Campaigns : TODO</div>
        </section>
      </Content>
    );
  }
}

CampaignsManager.propTypes = {
  appSetTitle: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  const { admin } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.loading;
  return { admin, isLoading, isSignedIn };
};

const mapDispatchToProps = dispatch => ({
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(CampaignsManager);
