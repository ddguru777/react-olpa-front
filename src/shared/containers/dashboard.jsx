import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Grid, Inner, Cell, Button } from "zrmc";
import Loading from "zoapp-front/components/loading";

import DonutChart from "../components/donutChart";
import { appSetTitle } from "../actions/app";

class Dashboard extends Component {
  componentWillMount() {
    this.props.appSetTitle("Dashboard");
  }

  render() {
    const infoStyle = {
      textAlign: "center",
      fontSize: "28px",
      fontWeight: "400",
      color: "#888",
      padding: "16px 0",
      lineHeight: "1.1",
    };
    const infoStyleB = {
      textAlign: "center",
      fontSize: "16px",
      fontWeight: "400",
      color: "#666",
      padding: "60px 0",
      lineHeight: "1.1",
    };
    const infoStyleC = {
      fontSize: "16px",
      fontWeight: "400",
      color: "green",
      padding: "60px 0",
      lineHeight: "1.1",
    };
    const infoStyleD = {
      fontSize: "16px",
      fontWeight: "400",
      color: "#666",
      padding: "16px",
      lineHeight: "1.1",
    };

    const { isLoading } = this.props;
    if (!this.props.isSignedIn) {
      return (<div>You need to sign in...</div>);
    } else if (isLoading || this.props.admin == null) {
      return (<Loading />);
    }
    const { admin } = this.props;
    const usersCount = admin.users != null ? admin.users.count : 0;
    const conversationsCount = admin.conversations != null ? admin.conversations.count : 0;
    const messagesCount = admin.messages != null ? admin.messages.count : 0;
    return (
      <div className="mdl-layout__content mdl-color--grey-100">
        <Grid>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <div style={infoStyleD}>
                Last 7 days |  Filter: All

                <Button
                  raised
                  style={{ float: "right", marginBottom: "16px" }}
                >
                  Export
                </Button>
                <Button
                  raised
                  style={{ float: "right", marginBottom: "16px", marginRight: "16px" }}
                >
                  Edit
                </Button>
              </div>

            </Cell>
          </Inner>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <Grid>
                <Inner>
                  <Cell span={4}>
                    <div style={infoStyle}>
                      <span style={infoStyle}>{usersCount}</span><br />
                      <span style={infoStyleB}>users / <span style={infoStyleC}>+4%</span></span>
                    </div>
                  </Cell>
                  <Cell span={4}>
                    <div style={infoStyle}>
                      <span style={infoStyle}>{conversationsCount}</span><br />
                      <span style={infoStyleB}>errors / <span style={infoStyleC}>+8%</span></span>
                    </div>
                  </Cell>
                  <Cell span={4}>
                    <div style={infoStyle}>
                      <span style={infoStyle}>{messagesCount}</span><br />
                      <span style={infoStyleB}>
                      messages /
                        <span style={infoStyleC}>+11%</span>
                      </span>
                    </div>
                  </Cell>
                </Inner>
              </Grid>
            </Cell>
          </Inner>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <Grid>
                <Inner>
                  <Cell span={3}>
                    <DonutChart title="Platform" dataset={0} />
                  </Cell>
                  <Cell span={3}>
                    <DonutChart title="Country" dataset={1} />
                  </Cell>
                  <Cell span={3}>
                    <DonutChart title="Language" dataset={2} />
                  </Cell>
                  <Cell span={3}>
                    <DonutChart title="Age" dataset={3} />
                  </Cell>
                </Inner>
              </Grid>
            </Cell>
          </Inner>
        </Grid>
      </div>
    );
  }
}

Dashboard.defaultProps = {
  admin: null,
  isLoading: false,
  isSignedIn: false,
};

Dashboard.propTypes = {
  admin: PropTypes.shape({ params: PropTypes.shape({}).isRequired }),
  isLoading: PropTypes.bool,
  isSignedIn: PropTypes.bool,
  appSetTitle: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  const { admin } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.loading;
  return { admin, isLoading, isSignedIn };
};

const mapDispatchToProps = dispatch => ({
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
