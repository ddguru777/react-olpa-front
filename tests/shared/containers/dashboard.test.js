/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import renderer from "react-test-renderer";

import { DashboardBase } from "shared/containers/dashboard";

describe("containers/Dashboard", () => {
  it("renders and update correctly", () => {
    const appSetTitleSpy = jest.fn();
    const fetchMetricsSpy = jest.fn();

    const metrics = {
      users: {
        count: 123,
      },
      conversations: {
        count: 1200,
        messages_per_conversation: 4,
      },
      sessions: {
        duration: 5000,
      },
      errors: {
        rate: 0.230115,
      },
      responses: {
        speed: 40,
      },
    };

    const component = renderer.create(
      <DashboardBase
        appSetTitle={appSetTitleSpy}
        fetchMetrics={fetchMetricsSpy}
        isSignedIn
        metrics={metrics}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(appSetTitleSpy).toHaveBeenCalled();
    expect(fetchMetricsSpy).not.toHaveBeenCalled();

    component.update(
      <DashboardBase
        appSetTitle={appSetTitleSpy}
        fetchMetrics={fetchMetricsSpy}
        isSignedIn
        selectedBotId={"bot1"}
        metrics={metrics}
      />,
    );
    expect(fetchMetricsSpy).toHaveBeenCalledWith("bot1");
  });

  it("renders a Loading component when user is not signed in", () => {
    const component = renderer.create(
      <DashboardBase
        appSetTitle={jest.fn()}
        fetchMetrics={jest.fn()}
        isSignedIn={false}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a Loading component when isLoading is true", () => {
    const component = renderer.create(
      <DashboardBase
        appSetTitle={jest.fn()}
        fetchMetrics={jest.fn()}
        isSignedIn
        isLoading
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a Loading component when no metrics", () => {
    const metrics = null;

    const component = renderer.create(
      <DashboardBase
        appSetTitle={jest.fn()}
        fetchMetrics={jest.fn()}
        isSignedIn
        metrics={metrics}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
