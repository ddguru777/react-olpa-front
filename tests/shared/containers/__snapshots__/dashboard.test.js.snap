// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Dashboard renders a Loading component when isLoading is true 1`] = `
<div
  className="mdc-card"
  style={
    Object {
      "margin": "auto",
      "width": "512px",
    }
  }
>
  <section
    className="mdc-card__primary"
  >
    
    <h1
      className="mdc-card__title"
    >
      Processing
    </h1>
    
  </section>
  <section
    className="mdc-card__supporting-text"
  >
    Please wait ...
  </section>
  <section
    className="mdc-card__actions"
  />
</div>
`;

exports[`containers/Dashboard renders a Loading component when no metrics 1`] = `
<div
  className="mdc-card"
  style={
    Object {
      "margin": "auto",
      "width": "512px",
    }
  }
>
  <section
    className="mdc-card__primary"
  >
    
    <h1
      className="mdc-card__title"
    >
      Processing
    </h1>
    
  </section>
  <section
    className="mdc-card__supporting-text"
  >
    Please wait ...
  </section>
  <section
    className="mdc-card__actions"
  />
</div>
`;

exports[`containers/Dashboard renders a Loading component when user is not signed in 1`] = `
<div>
  You need to sign in...
</div>
`;

exports[`containers/Dashboard renders correctly 1`] = `
<div
  className="mdl-layout__content mdl-color--grey-100"
>
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdl-color--white mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
      >
        <div
          className="mdc-layout-grid"
        >
          <div
            className="mdc-layout-grid__inner"
          >
            <div
              className="mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
            >
              <h2>
                Key metrics
              </h2>
            </div>
          </div>
          <div
            className="mdc-layout-grid__inner"
          >
            <div
              className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
            >
              <div
                style={
                  Object {
                    "margin": "1em",
                    "textAlign": "center",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#888",
                      "fontSize": "34px",
                      "fontWeight": "500",
                      "lineHeight": "1.1",
                      "padding": "16px 0",
                    }
                  }
                >
                  123
                </span>
                <br />
                <span
                  style={
                    Object {
                      "color": "#666",
                      "fontSize": "16px",
                      "fontWeight": "400",
                      "lineHeight": "1.1",
                      "padding": "60px 0",
                      "textAlign": "center",
                    }
                  }
                >
                  users
                </span>
              </div>
            </div>
            <div
              className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
            >
              <div
                style={
                  Object {
                    "margin": "1em",
                    "textAlign": "center",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#888",
                      "fontSize": "34px",
                      "fontWeight": "500",
                      "lineHeight": "1.1",
                      "padding": "16px 0",
                    }
                  }
                >
                  1200
                </span>
                <br />
                <span
                  style={
                    Object {
                      "color": "#666",
                      "fontSize": "16px",
                      "fontWeight": "400",
                      "lineHeight": "1.1",
                      "padding": "60px 0",
                      "textAlign": "center",
                    }
                  }
                >
                  conversations
                </span>
              </div>
            </div>
            <div
              className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
            >
              <div
                style={
                  Object {
                    "margin": "1em",
                    "textAlign": "center",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#888",
                      "fontSize": "34px",
                      "fontWeight": "500",
                      "lineHeight": "1.1",
                      "padding": "16px 0",
                    }
                  }
                >
                  4
                </span>
                <br />
                <span
                  style={
                    Object {
                      "color": "#666",
                      "fontSize": "16px",
                      "fontWeight": "400",
                      "lineHeight": "1.1",
                      "padding": "60px 0",
                      "textAlign": "center",
                    }
                  }
                >
                  messages/conversations
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdl-color--white mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
      >
        <div
          className="mdc-layout-grid"
        >
          <div
            className="mdc-layout-grid__inner"
          >
            <div
              className="mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
            >
              <h2>
                Platform metrics
              </h2>
            </div>
          </div>
          <div
            className="mdc-layout-grid__inner"
          >
            <div
              className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
            >
              <div
                style={
                  Object {
                    "margin": "1em",
                    "textAlign": "center",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#888",
                      "fontSize": "34px",
                      "fontWeight": "500",
                      "lineHeight": "1.1",
                      "padding": "16px 0",
                    }
                  }
                >
                  5000
                  ms
                </span>
                <br />
                <span
                  style={
                    Object {
                      "color": "#666",
                      "fontSize": "16px",
                      "fontWeight": "400",
                      "lineHeight": "1.1",
                      "padding": "60px 0",
                      "textAlign": "center",
                    }
                  }
                >
                  session duration
                </span>
              </div>
            </div>
            <div
              className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
            >
              <div
                style={
                  Object {
                    "margin": "1em",
                    "textAlign": "center",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#888",
                      "fontSize": "34px",
                      "fontWeight": "500",
                      "lineHeight": "1.1",
                      "padding": "16px 0",
                    }
                  }
                >
                  23
                  %
                </span>
                <br />
                <span
                  style={
                    Object {
                      "color": "#666",
                      "fontSize": "16px",
                      "fontWeight": "400",
                      "lineHeight": "1.1",
                      "padding": "60px 0",
                      "textAlign": "center",
                    }
                  }
                >
                  errors rate
                </span>
              </div>
            </div>
            <div
              className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
            >
              <div
                style={
                  Object {
                    "margin": "1em",
                    "textAlign": "center",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#888",
                      "fontSize": "34px",
                      "fontWeight": "500",
                      "lineHeight": "1.1",
                      "padding": "16px 0",
                    }
                  }
                >
                  40
                  ms
                </span>
                <br />
                <span
                  style={
                    Object {
                      "color": "#666",
                      "fontSize": "16px",
                      "fontWeight": "400",
                      "lineHeight": "1.1",
                      "padding": "60px 0",
                      "textAlign": "center",
                    }
                  }
                >
                  response time
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
